File: backend/app/schemas/jobs.py
from pydantic import BaseModel, ConfigDict


class JobCreate(BaseModel):
    group_id: str
    reference: str


class JobOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    tenant_id: str
    group_id: str
    reference: str
    status: str


class JobDocumentOut(BaseModel):
    id: str
    template_document_id: str
    name: str
    doc_type: str
    is_uploaded: bool
    page_count: int


class JobFieldValueOut(BaseModel):
    id: str
    mark_id: str
    template_document_id: str
    document_name: str
    label_name: str
    extracted_value: str | None
    corrected_value: str | None
    value: str | None


class VerificationRow(BaseModel):
    link_id: str
    field_label: str
    source_document: str
    source_value: str | None
    target_document: str
    target_value: str | None
    status: str  # match | mismatch | missing | review
    accepted: bool = False


class VerificationDecision(BaseModel):
    link_id: str
    accept: bool  # True = accept this row, False = un-accept


class JobDetailOut(BaseModel):
    id: str
    tenant_id: str
    group_id: str
    group_name: str
    reference: str
    status: str
    documents: list[JobDocumentOut]
    field_values: list[JobFieldValueOut]
    verifications: list[VerificationRow]
    all_checks_passed: bool


class FieldValueCorrect(BaseModel):
    corrected_value: str


class AvailableGroup(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    tenant_id: str
    name: str
    status: str

File: frontend/src/types/jobs.ts
export interface Job {
  id: string;
  tenant_id: string;
  group_id: string;
  reference: string;
  status: string;
}

export interface JobDocument {
  id: string;
  template_document_id: string;
  name: string;
  doc_type: string;
  is_uploaded: boolean;
  page_count: number;
}

export interface JobFieldValue {
  id: string;
  mark_id: string;
  template_document_id: string;
  document_name: string;
  label_name: string;
  extracted_value: string | null;
  corrected_value: string | null;
  value: string | null;
}

export type VerificationStatus = "match" | "mismatch" | "missing" | "review";

export interface VerificationRow {
  link_id: string;
  field_label: string;
  source_document: string;
  source_value: string | null;
  target_document: string;
  target_value: string | null;
  status: VerificationStatus;
  accepted: boolean;
}

export interface JobDetail {
  id: string;
  tenant_id: string;
  group_id: string;
  group_name: string;
  reference: string;
  status: string;
  documents: JobDocument[];
  field_values: JobFieldValue[];
  verifications: VerificationRow[];
  all_checks_passed: boolean;
}
